ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/PropertyOverrider.java

```diff
@@ -18,29 +18,49 @@
 import com.google.common.collect.ImmutableMap;
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import io.confluent.ksql.KsqlExecutionContext;
-import io.confluent.ksql.config.KsqlConfigResolver;
 import io.confluent.ksql.parser.tree.SetProperty;
 import io.confluent.ksql.parser.tree.UnsetProperty;
+import io.confluent.ksql.rest.client.properties.LocalPropertyValidator;
+import io.confluent.ksql.rest.entity.KsqlEntity;
 import io.confluent.ksql.services.ServiceContext;
 import io.confluent.ksql.statement.ConfiguredStatement;
 import io.confluent.ksql.util.KsqlStatementException;
+import java.util.Optional;
 
-public final class PropertyValidator {
+public final class PropertyOverrider {
 
-  private PropertyValidator() { }
+  private PropertyOverrider() { }
 
-  @SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_INFERRED")
-  public static void set(
-      final ConfiguredStatement<?> statement,
+  public static Optional<KsqlEntity> set(
+      final ConfiguredStatement<SetProperty> statement,
       final KsqlExecutionContext context,
       final ServiceContext serviceContext
   ) {
-    final SetProperty setProperty = (SetProperty) statement.getStatement();
-    throwIfUnknownProperty(
-        setProperty.getPropertyName(),
-        statement.getStatementText()
-    );
+    if (statement.getStatement() != null) {
```

Comment:
Why would this be `null`? Can it even be null? I think all the Pojo's being called throw NPEs on construction to protect against this.

Reply:
I think it might be mocks. I'll figure this out and clean it up.